### DISCUSSION TITLE
How do you discard changes in one file?

### DISCUSSION THREAD

**Discussion by metov** [jesseduffield/lazygit#2154] [Q&A] [answered]
Say I have a lot of unstaged changes in a repo. I have a file or subdir selected in the `Files` window. How do I do the equivalent of `git reset foo/bar/` or `git checkout foo/bar/` so I can throw away changes on that file/dir only? I've seen the `d` and `D` options but they all seem to apply to the whole directory, not just a path.

**Comment by mark2185** [comment, marked as answer]
If you want to discard just a specific file, move the cursor to the file.

`d` discard whatever is under the cursor, it's as if you wrote `git checkout -- <thing under cursor>`. If it's a directory, it'll discard that directory. If it's a file, it'll discard that file.

**Reply by sebkolind** [reply]
Ah, I thought `Discard all changes` meant that it would discard all unstaged changes. Then I thought that maybe it should be reworded, but then I realised that the "header" of the popup window includes the specific file that you are doing the discard operation on.

**Reply by metov** [reply]
That's what I thought as well, which is why I asked the question. I'm not illiterate :D But the option is worded as if it will nuke my entire working tree instead of just the one file.

IMO it should really be renamed to say `discard changes in file` because that's what it actually does.

**Reply by mark2185** [reply]
That is stated in the current `master`. There's a little tooltip below the menu window that says "Discard both staged and unstaged changes in <file>."

**Reply by markojak** [reply]
haha I just found this thread because I was too scared to select "Discard all changes"
